File: SecondYear/gamandeepsingh_GamandeepSingh_2226csit1042_2/Week_3/SortingThreeGroups.java
class Solution {
    public int minimumOperations(List<Integer> nums) {
        int i=0, n=nums.size(), res=Integer.MAX_VALUE;
        int arr[][]=new int[n][4];
        for(int num: nums) arr[i++][num]++;
        for(int f=1;f<n;f++){
            arr[f][1]+=arr[f-1][1];
            arr[f][2]+=arr[f-1][2];
            arr[f][3]+=arr[f-1][3];
        }
        for(i=-1;i<n;i++){
            int toone=0;
            if(i>=0) toone=arr[i][2]+arr[i][3];
            for(int j=i;j<n;j++){
                int totwo=0, tothree=0;
                if(j>=0&&i<n-1)
                    totwo=(arr[j][1]-(-1==i?0:arr[i][1]))+(arr[j][3]-(-1==i?0:arr[i][3]));
                if(i<n-2)
                    tothree=(arr[n-1][1]-(j==-1?0:arr[j][1]))+(arr[n-1][2]-(j==-1?0:arr[j][2]));
                res=Math.min(res, toone + totwo + tothree);
            }
        }
        return Math.min(res, arr[n-1][1]+arr[n-1][2]);
    }
}

File: SecondYear/any-mesh_AnimeshKumarSingh_2226CSE(AI&ML)1119_2/Week_3/sortingThreeGroups.cpp
class Solution {
private:
    int func(vector<int>& nums, int n, int j, int k) {
        int res = 0;
        for(int i = 0; i < j && i < n; i++) {
            if(nums[i] != 1) {
                res++;
            }
        }
        for(int i = j; i < k && i < n; i++) {
            if(nums[i] != 2) {
                res++;
            }
        }
        for(int i = k; i < n; i++) {
            if(nums[i] != 3) {
                res++;
            }
        }
        return res;
    }
public:
    int minimumOperations(vector<int>& nums) {
        if(is_sorted(nums.begin(), nums.end())) {
            return 0;
        }
        int result = INT_MAX, n = nums.size();
        for(int k = n + 1; k >= 0; k--) {
            for(int j = k; j >= 0; j--) {
                result = min(result, func(nums, n, j, k));
            }
        }
        return result;
    }
};
